aws-amplify/amplify-js
# feat(auth): handles DEVICE_SRP_AUTH challengeName

#### Description of changes

This PR handles the DEVICE_SRP_AUTH challengeName that is returned when a DEVICE is remembered in the Cognito User Pool. This change will allow to bypass MFA if this option was previously enabled in the user pool.

#### Issue #, if available




#### Description of how you validated changes

Tested the following steps in a sample app:

1. Enabled `Always remember` on the `Device Tracking` section of the Cognito UserPool console.
2. Enabled `Require MFA`
3. Login using any `authFlow` (SRP, CUSTOM, etc)
4. Handle MFA 
5. Log out
6. Login again.
7. User should be authenticated without having to address MFA.


#### Checklist



By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
